### PR TITLE
add cluster-sync periodic job for s390x.

### PR DESCRIPTION
**Pull request by vamsikrishna-siddu**
Hi @brianmcarey  
   I have tested this in my local and this test is passing. Can you please take a look at it when you get some time and let me know your thoughts?

### CODE REVIEW

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-periodics.yaml [hunk 1]
```diff
@@ -1127,6 +1127,96 @@ periodics:
...
+          - "-c"
+          - |
+            # install yq
+            curl -Lo ./yq https://github.com/mikefarah/yq/releases/download/3.4.1/yq_linux_amd64
+            chmod +x ./yq && mv ./yq /usr/local/bin/yq
+
+            # get kubeconfig
+            source ../project-infra/hack/manage-secrets.sh
+            decrypt_secrets
+            extract_secret 'kubeconfigS390X' /kubeconfig
```
Comment (brianmcarey) [inline github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-periodics.yaml:1194]: this secret doesn't exist  - which external cluster is this job supposed to target?

Comment (vamsikrishna-siddu) [reply github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-periodics.yaml:1194]: yeah you are correct. Can you please create a secret for the external cluster s390x-prow-virtual cluster and add it with the name kubeconfigs390x. I want this job to target the external 390x-prow-virtual cluster.

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml [hunk 2]
```diff
@@ -1043,6 +1043,95 @@ presubmits:
...
+    annotations:
+      fork-per-release: "true"
+      testgrid-dashboards: kubevirt-presubmits
+    cluster: prow-workloads
+    decorate: true
+    decoration_config:
+      timeout: 4h0ms
+      grace_period: 5m0s
+    labels:
+      preset-dind-enabled: "true"
```
Comment (brianmcarey) [inline github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml:1056]:
```suggestion
      preset-podman-in-container-enabled: "true"
```

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml [hunk 3]
```diff
@@ -1043,6 +1043,95 @@ presubmits:
...
         name: audit
+  - always_run: false
+    annotations:
+      fork-per-release: "true"
+      testgrid-dashboards: kubevirt-presubmits
+    cluster: prow-workloads
+    decorate: true
+    decoration_config:
+      timeout: 4h0ms
+      grace_period: 5m0s
```
Comment (brianmcarey) [inline github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml:1054]: Need to add in order to handle getting the kubeconfig
```suggestion
    extra_refs:
    - base_ref: main
      org: kubevirt
      repo: project-infra
```

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml [hunk 4]
```diff
@@ -1043,6 +1043,95 @@ presubmits:
           path: /var/log/audit
           type: Directory
         name: audit
+  - always_run: false
+    annotations:
+      fork-per-release: "true"
+      testgrid-dashboards: kubevirt-presubmits
+    cluster: prow-workloads
```
Comment (brianmcarey) [inline github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml:1050]:
```suggestion
    cluster: kubevirt-prow-workloads
```

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml [hunk 5]
```diff
@@ -1043,6 +1043,98 @@ presubmits:
...
+    extra_refs:
+    - base_ref: main
+      org: kubevirt
+      repo: project-infra
+    labels:
+      preset-podman-in-container-enabled: "true"
+      preset-docker-mirror-proxy: "true"
+      preset-shared-images: "true"
+      preset-bazel-cache: "true"
+      preset-bazel-unnested: "true"
```
Comment (brianmcarey) [inline github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml:1063]: This preset will have to be dropped as we require some of the packages that only present in the kubevirt builder image

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-periodics.yaml (hunk 1 above)
Comment (vamsikrishna-siddu) [reply github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-periodics.yaml]: resolving this conversation as the secret get successfully added with the name `kubeconfigS390X` here.
https://github.com/kubevirt/project-infra/issues/3540

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml (hunk 5 above)
Comment (vamsikrishna-siddu) [reply github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml]: Hi @brianmcarey  I have removed the preset `preset-bazel-unnested: "true"`

**Comment by brianmcarey** [comment]
@vamsikrishna-siddu those this need to be converted back to a periodic test lane or do you want to add it as a presubmit test lane?

**Comment by vamsikrishna-siddu** [comment]
> @vamsikrishna-siddu does this need to be converted back to a periodic test lane or do you want to add it as a presubmit test lane?

Hi @brianmcarey  I want to add it as a presubmit test lane

**Comment by brianmcarey** [comment]
> > @vamsikrishna-siddu does this need to be converted back to a periodic test lane or do you want to add it as a presubmit test lane?
> 
> Hi @brianmcarey I want to add it as a presubmit test lane

As this job is using some sensitive secrets - I would prefer to run this as a periodic - maybe 4 times a day to get some good coverage?

**Comment by vamsikrishna-siddu** [comment]
> > > @vamsikrishna-siddu does this need to be converted back to a periodic test lane or do you want to add it as a presubmit test lane?
> > 
> > 
> > Hi @brianmcarey I want to add it as a presubmit test lane
> 
> As this job is using some sensitive secrets - I would prefer to run this as a periodic - maybe 4 times a day to get some good coverage?

sure @brianmcarey

**Review by brianmcarey — COMMENTED** [review]
@vamsikrishna-siddu  just to confirm - this is no longer needed right?

**Comment by vamsikrishna-siddu** [comment]
Hi @brianmcarey ,
Yes it is not needed. I can create this job again once we get the provider for s390x is ready.

**Pull request closed without merging**